sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -76,6 +76,22 @@ private static AmqpMessage BuildAmqpBatchFromMessage(IReadOnlyCollection<Service
             return BuildAmqpBatchFromMessages(batchMessages, firstMessage, forceBatch);
         }
 
+        /// <summary>
+        ///   Builds a batch of <see cref="AmqpMessage" /> from a set of <see cref="ServiceBusMessage" />
+        ///   optionally propagating the custom properties.
+        /// </summary>
+        ///
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
+        /// <param name="firstMessage">The first message being sent in the batch.</param>
+        /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
+        ///
+        /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
+        ///
+        public virtual AmqpMessage BuildAmqpBatchFromMessages(IReadOnlyCollection<AmqpMessage> source, ServiceBusMessage firstMessage, bool forceBatch)
+        {
+            return BuildAmqpBatchFromMessages(source.ToList<AmqpMessage>(), firstMessage, forceBatch);
```

Comment:
I don't think we need to force list here and eat the allocation.   Can we either update the parameter to use `IReadOnlyCollection<AmqpMessage>` here?